### PR TITLE
Add mailto link for user email at order details page

### PR DESCRIPTION
**Pull request by mstanielewicz**
Close https://github.com/mirumee/saleor/issues/2869

### Pull Request Checklist

1. [ ] Privileged views and APIs are guarded by proper permission checks.
1. [ ] All visible strings are translated with proper context.
1. [ ] All data-formatting is locale-aware (dates, numbers, and so on).
1. [ ] Database queries are optimized and the number of queries is constant.
1. [ ] The changes are tested.
1. [ ] The code is documented (docstrings, project documentation).
1. [ ] GraphQL schema and type definitions are up to date.

### CODE REVIEW

File: saleor/static/dashboard-next/storybook/__snapshots__/Stories.test.ts.snap [hunk 1]
```diff
@@ -31084,7 +31084,7 @@ exports[`Storyshots Views / Products / Product list default 1`] = `
             <td
               class="MuiTableCell-root-id MuiTableCell-body-id Component-textRight-id"
             >
-              XAG 274.99
+              XAG274.99
```
Comment (dominik-zeglen) [inline saleor/static/dashboard-next/storybook/__snapshots__/Stories.test.ts.snap]: Looks like you're using earlier than 10 version of node

File: saleor/static/dashboard-next/orders/mutations.ts [hunk 2]
```diff
@@ -13,8 +13,8 @@ import { OrderRelease, OrderReleaseVariables } from "./types/OrderRelease";
 
 const orderCancelMutation = gql`
   ${fragmentOrderDetails}
-  mutation OrderCancel($id: ID!) {
-    orderCancel(id: $id, restock: true) {
+  mutation OrderCancel($id: ID!, $restock: Boolean!) {
```
Comment (dominik-zeglen) [inline saleor/static/dashboard-next/orders/mutations.ts]: This does not belong here

File: saleor/static/dashboard-next/orders/components/OrderCustomer/OrderCustomer.tsx [hunk 3]
```diff
@@ -93,17 +91,12 @@ const OrderCustomer = decorate<OrderCustomerProps>(
...
               className={classes.link}
-              onClick={
-                onCustomerEmailClick
-                  ? onCustomerEmailClick(client.id)
-                  : undefined
-              }
             >
-              {client.email}
-            </Typography>
+              <Typography>{client.email}</Typography>
```
Comment (dominik-zeglen) [inline saleor/static/dashboard-next/orders/components/OrderCustomer/OrderCustomer.tsx]: Can we refactor it to `<Typography component="a" href="..." />`?
I feel like it should be done that way.

File: saleor/static/dashboard-next/storybook/__snapshots__/Stories.test.ts.snap (hunk 1 above)
Comment (mstanielewicz) [reply saleor/static/dashboard-next/storybook/__snapshots__/Stories.test.ts.snap]: Updated node and snapshots.

File: saleor/static/dashboard-next/orders/mutations.ts (hunk 2 above)
Comment (mstanielewicz) [reply saleor/static/dashboard-next/orders/mutations.ts]: Removed.

File: saleor/static/dashboard-next/orders/components/OrderCustomer/OrderCustomer.tsx (hunk 3 above)
Comment (mstanielewicz) [reply saleor/static/dashboard-next/orders/components/OrderCustomer/OrderCustomer.tsx]: Couldn't add href prop to <Typography />.
Created <ExternalLink />.

**Pull request merged**